packages/editor/src/browser/editor-navigation-contribution.ts

```diff
@@ -83,7 +84,17 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
             isEnabled: () => !!this.locationStack.lastEditLocation()
         });
         this.commandRegistry.registerHandler(EditorCommands.CLEAR_EDITOR_HISTORY.id, {
-            execute: () => this.locationStack.clearHistory(),
+            execute: async () => {
+                const confirmed = await new ConfirmDialog({
```

Comment:
minor: we should probably rename in the context of the action

```suggestion
                const shouldClear = await new ConfirmDialog({
```